### PR TITLE
Ensure that ClientConnectionStrategyConfig is not overwritten while parsing

### PR DESCRIPTION
**Pull request by pivovarit**

### CODE REVIEW

**Comment by mmedenjak** [comment]
Moving out of 4.2. If you manage to get it reviewed and merged before release branch is created, feel free to move back.

**Comment by pivovarit** [comment]
To be honest, I thought it was merged a long time ago. Let's see if I can get it reviewed soon enough.

File: hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalClientConfigurationOverrideTest.java [hunk 1]
```diff
@@ -42,4 +44,18 @@ public void shouldExtractConfigFromEnv() throws Exception {
         assertEquals("test", config.getInstanceName());
         assertFalse(config.getNetworkConfig().isAutoDetectionEnabled());
     }
+
+    @Test
+    public void shouldHandleConnectionStrategy() throws Exception {
+        ClientConfig config = new ClientConfig();
+        config.getConnectionStrategyConfig()
+          .setReconnectMode(ASYNC)
+          .setAsyncStart(true);
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalClientConfigurationOverrideTest.java:53]: Shouldn't this be `false` to validate if the override is effective?

Comment (pivovarit) [reply hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalClientConfigurationOverrideTest.java:53]: Good point. However, the most important part here is the validation of the part that's not supposed to be overwritten. But yeah, will change it

**Review by blazember — APPROVED** [review]

**Comment by mmedenjak** [comment]
@pivovarit is the thing missing the API team review? Or anything else?

**Comment by pivovarit** [comment]
@mmedenjak technically it was just the review from the team

**Review by sancar — APPROVED** [review]

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
@pivovarit for some reason checkstyle chokes on this:
```
18:23:45 Caused by: java.lang.IllegalStateException: /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalClientConfigurationOverrideTest.java:61:12: expecting "class", found 'void'
```

Can you check it out?

**Comment by pivovarit** [comment]
Interesting... will try to check it out today

**Comment by mmedenjak** [comment]
@pivovarit can we merge this?

**Pull request merged**